scheds/rust/scx_chaos/src/bpf/main.bpf.c

```diff
@@ -457,15 +463,38 @@ void BPF_STRUCT_OPS(chaos_runnable, struct task_struct *p, u64 enq_flags)
 	return p2dq_runnable_impl(p, enq_flags);
 }
 
+void BPF_STRUCT_OPS(chaos_running, struct task_struct *p)
+{
+	struct chaos_task_ctx *taskc;
+
+	if (!(taskc = lookup_create_chaos_task_ctx(p))) {
+		scx_bpf_error("failed to lookup task context in enqueue");
+		return;
+	}
+
+	p2dq_running_impl(p);
+
+	s32 task_cpu = scx_bpf_task_cpu(p);
+	if (taskc->next_trait == CHAOS_TRAIT_CPU_FREQ) {
+		if (cpu_freq_min > 0) {
+			dbg("chaos freq pid: %d freq: %d", p->pid, cpu_freq_min);
+			scx_bpf_cpuperf_set(task_cpu, cpu_freq_min);
+		}
+	} else {
+		if (cpu_freq_max > 0)
+			scx_bpf_cpuperf_set(task_cpu, cpu_freq_max);
+	}
```

Comment:
Coming back to the last comment, I wonder if we should do the reset first? So:

```
if (taskc->next_trait != CHAOS_TRAIT_CPU_FREQ && cpu_freq_max > 0)
  scx_bpf_cpuperf_set(task_cpu, cpu_freq_max);

p2dq_running_impl(p);

if (taskc->next_trait == CHAOS_TRAIT_CPU_FREQ && cpu_freq_min > 0)
  scx_bpf_cpuperf_set(task_cpu, cpu_freq_min);
```

Not entirely sure how that would interact with p2dq though if it expects to set the value once and have it stick.